injection/sharedmain/main.go

```diff
@@ -128,10 +132,18 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 		controllers = append(controllers, cf(ctx, cmw))
 	}
 
+	profilingHandler := profiling.NewHandler(false)
+
 	// Watch the logging config map and dynamically update logging levels.
 	cmw.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
-	// Watch the observability config map and dynamically update metrics exporter.
-	cmw.Watch(metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger))
+	// Watch the observability config map
+	for _, observer := range []func(*corev1.ConfigMap){
+		metrics.UpdateExporterFromConfigMap(component, logger),
+		profiling.UpdateProfilingFromConfigMap(profilingHandler, logger),
+	} {
+		cmw.Watch(metrics.ConfigMapName(), observer)
+	}
```

Comment:
When I commented on this earlier, I kinda envisioned it to look more like this:

```golang
configMapWatcher.Watch(metrics.ConfigMapName(), func(cm *corev1.ConfigMap) {
	metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger)(cm)
	updateRequestLogFromConfigMap(logger, reqLogHandler)(cm)
})
```

The extra Watch calls might not actually hurt, to me they feel more expensive though. For posterity: I proposed to collapse these multiple Watch statements into one to cut down some repitition.